### PR TITLE
Add abstract param layout formatter test

### PR DESCRIPTION
**Pull request by leaf-soba**
- add unit test and refactor AbstractParamLayoutFormatter.parseArgument
- reason:
1. origin parseArgument method use too many if-else to make it confusing.
2. I want to refactor it but it don't have any unit test.
- screenshot:

### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/test/java/org/jabref/logic/layout/format/AbstractParamLayoutFormatterTest.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class AbstractParamLayoutFormatterTest {
+    static class ParseArgumentTester extends AbstractParamLayoutFormatter {
+        public static List<String> callParseArgument(String arg) {
+            return parseArgument(arg);
+        }
+
+        @Override
+        public String format(String fieldText) {
```
Comment (InAnYan) [inline src/test/java/org/jabref/logic/layout/format/AbstractParamLayoutFormatterTest.java:18]: Why `String format(String)` and `void setArgument(String)` needs to be overrided?

Comment (InAnYan) [reply src/test/java/org/jabref/logic/layout/format/AbstractParamLayoutFormatterTest.java:18]: Ah, I see, the interface requires `void setArgument(String arg)`. But what about `format`?

**Review by InAnYan — COMMENTED** [review]
Great code! Now it's much easier to understand the algorithm of the parser

File: src/test/java/org/jabref/logic/layout/format/AbstractParamLayoutFormatterTest.java (hunk 1 above)
Comment (leaf-soba) [reply src/test/java/org/jabref/logic/layout/format/AbstractParamLayoutFormatterTest.java:18]: Same reason to `setArgument`. The interface `ParamLayoutFormatter` extends `LayoutFormatter`, which is also an interface requires `format` method.

**Review by InAnYan — APPROVED** [review]

**Review by Siedlerchr — APPROVED** [review]

**Pull request merged**